imcsdk/apis/admin/ldap.py

```diff
@@ -0,0 +1,549 @@
+# Copyright 2016 Cisco Systems, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+This module implements apis to configure ldap
+"""
+
+import logging
+from imcsdk.imcexception import ImcOperationError
+from imcsdk.apis.utils import _get_mo, _is_valid_arg, _is_invalid_value
+from imcsdk.mometa.aaa.AaaLdap import AaaLdap
+from imcsdk.mometa.aaa.AaaLdapRoleGroup import AaaLdapRoleGroup
+
+log = logging.getLogger('imc')
+
+_LDAP_SERVERS = ['ldap_server1', 'ldap_server2', 'ldap_server3',
+                 'ldap_server4', 'ldap_server5', 'ldap_server6']
+_LDAP_SERVER_PORTS = ['ldap_server_port1', 'ldap_server_port2',
+                      'ldap_server_port3', 'ldap_server_port4',
+                      'ldap_server_port5', 'ldap_server_port6']
+LDAP_DN = "sys/ldap-ext"
+
+
+def _set_ldap_servers(mo, ldap_servers):
+    if len(ldap_servers) > len(_LDAP_SERVERS):
+        raise ImcOperationError("Configure LDAP", "Cannot configure more than"
+                                "%d servers" % len(_LDAP_SERVERS))
+    servers = {}
+    ports = {}
+    for server in ldap_servers:
+        servers['ldap_server' + str(server['id'])] = server['ip']
+        ports['ldap_server_port' + str(server['id'])] = str(server['port'])
```

Comment:
The below compresses you 5 lines into 2
servers = {'ldap_server'+str(x['id']): x['ip'] for x in ldap_servers}
ports = {'ldap_server_port'+str(x['id']): x['port'] for x in ldap_servers}